dp-identity-data-migration/userdataextraction/user_extraction.go

```diff
@@ -144,10 +170,54 @@ func process_zebedee_users(csvwriter *csv.Writer, userlist []zebedee.User) {
 		csvline.mfa_enabled = "FALSE"
 		csvline.phone_number_verified = "FALSE"
 		csvline.email_verified = "TRUE"
-		if err := csvwriter.Write(convert_to_slice(csvline)); err != nil {
-			fmt.Println("error writing record to csv:", err)
+
+		if isValidEmail := validateEmailId(validEmailDomains, user.Email); isValidEmail {
+			if err := validUsersWriter.Write(convert_to_slice(csvline)); err != nil {
+				fmt.Println("error writing record to csv:", err)
+			}else {
+				validUsersCount += 1
+			}
+		} else {
+			if err := invalidUsersWriter.Write(convert_to_slice(csvline)); err != nil {
```

Comment:
`convert_to_slice(csvline)` is run no matter what (in both sides of the if) so maybe consider moving it up to line 173 and use a variable?